File: Week4/netcore-automation-test/pages/HomePage.cs
using System;
using OpenQA.Selenium;


namespace netcore_automation_test
{

    public class HomePage : Methods
    {
        public IWebDriver webDriver;
        public HomePage(IWebDriver driver) : base(driver)
        {
            webDriver = driver;
        }

        public IWebElement Cookie => webDriver.FindElement(By.ClassName("a-button-primary"));
        public IWebElement AccountandList => webDriver.FindElement(By.Id("nav-link-accountList"));
        public IWebElement LoginButton => webDriver.FindElement(By.CssSelector("#nav-flyout-ya-signin .nav-action-inner"));
        public IWebElement AccountName => webDriver.FindElement(By.Id("nav-link-accountList-nav-line-1"));
        public IWebElement ElectronicMenu => webDriver.FindElement(By.XPath("//a[text()= 'Elektronik']"));

        public LoginPage GetLoginPage()
        {
            ClickElement(Cookie);
            HoverElement(AccountandList);
            ClickElement(LoginButton);
            return new LoginPage(webDriver);
        }

        public String GetAccountName()
        {
            return GetTextElement(AccountName);
        }
        public ProductsPage SelectElectronicMenu()
        {
            ClickElement(ElectronicMenu);
            return new ProductsPage(webDriver);
        }

    }


}

File: Week4/python-automation-test/pages/home_page.py
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By


class HomePage():
    def __init__(self, driver):
        self.driver = driver

        self.cookie_class_name = "a-button-primary"
        self.account_list_id = "nav-link-accountList"
        self.login_button_css = "#nav-flyout-ya-signin .nav-action-inner"
        self.electronic_menu_xpath = "//a[text()= 'Elektronik']"

    def get_login_page(self):
        self.driver.find_element(By.CLASS_NAME, self.cookie_class_name).click()
        element = self.driver.find_element(By.ID, self.account_list_id)
        actions = ActionChains(self.driver)
        actions.move_to_element(element).perform()
        self.driver.find_element(By.CSS_SELECTOR, self.login_button_css).click()

    def select_electronic_menu(self):
        self.driver.find_element(By.XPATH, self.electronic_menu_xpath).click()